sycl/source/device_selector.cpp

```diff
@@ -157,5 +161,152 @@ int host_selector::operator()(const device &dev) const {
   return Score;
 }
 
+namespace ONEAPI {
+namespace detail {
+struct filter {
+  backend Backend;
+  RT::PiDeviceType DeviceType;
```

Comment:
Please, default initialize to, for example, all device types.

Reply:
Ok.  Any particular reason?

Reply:
Usually default initialization helps when debugging - if this variable is not initialized later you are getting defined value rather then random garbage each time you run an app.